deployer.py

```diff
@@ -110,6 +110,31 @@ def stop_bot(bot_name):
     for container in containers:
         for tag in container.image.tags:
             if tag.startswith(bot_name.replace('@', '')):
+                logs = container.logs().decode("utf-8")
+                with open('bots/' + bot_name + '/logs.txt', 'a') as logfile:
+                    logfile.write("Container id: " + container.short_id + "\n")
+                    logfile.write("Stop Time: " + str(datetime.now()) + "\n")
+                    logfile.write(logs + "\n")
+                    logfile.write("--------------------\n")
                 container.stop()
                 return True
     return False
+
+def bot_log(bot_name, **kwargs):
+    lines = kwargs.get('lines', None)
+    if lines is not None:
+        lines = int(lines)
+    containers = docker_client.containers.list(all=True)
+    for container in containers:
+        for tag in container.image.tags:
+            if tag.startswith(bot_name.replace('@', '')):
+                logs = container.logs().decode("utf-8")
+                if lines is None:
+                    return logs
+                else:
+                    log_data = ''
+                    logs = logs.split('\n')
+                    for log in logs[:lines]:
```

Comment:
How about `return logs[lines:].join('\n')`? We want the last 'n' lines of logs here, not the first.

Also, since there's a possibility that the num of lines requested is more than the complete log size, so we should essentially do this, but in a prettier way:

```py
lines = max(0, len(logs)-lines)
return '\n'.join(logs.split('\n')[lines:])
```

Reply:
`return logs[lines:].join('\n')` doesn't work because join is a string only operation and earlier I tried returning only `return logs[:lines]` but that gives and error `TypeError: 'list' object is not callable` so only returning `return logs[lines:]` also doesn't works.

Reply:
In the sample code also, I wrote `str.join(list/callable)`. Yes, joins work kind of reverse of how we expect them to coming from JS, but they are there.